core/ledger/ledger_interface.go

```diff
@@ -700,6 +700,14 @@ type HashProvider interface {
 	GetHash(opts bccsp.HashOpts) (hash.Hash, error)
 }
 
+// ChannelInfoProvider interface enables ledger components to retrieve channel config data in all the config
+// blocks (current and previous config blocks).
+type ChannelInfoProvider interface {
```

Comment:
This interface will be used by privacyenabledstate.NewDBProvider. The  NewDBProvider method will call the interface to get MSPIDs and pass MSPIDs to DeployedChaincodeInfoProvider to construct implicit collection names. The interface is not directly used by DeployedChaincodeInfoProvider because it serves a special purpose and it cannot be instantiated until ledger is initialized. 

The interface is needed to avoid circular dependency and the mock is needed for unit tests. It will be used by https://jira.hyperledger.org/browse/FAB-17795.

Reply:
IMO, we can make kv_ledger_provider.go call these APIs (only if the ledgerID exists already) and pass mspIDs to the required pkg (here, privacyenabledstate pkg). `ImplicitCollectionNames(mspIDs []string) []string` in `DeployedInfoCCProvider` (to be added) can be used by privacyenabledstate pkg to fetch the actual collection name.

Reply:
Discussed at scrum. The interface will be used by privacyenabledstate.NewDBProvider.